### PR TITLE
Support go mod, new makefile and some logs issues

### PR DESCRIPTION
**Pull request by insidieux**

### CODE REVIEW

**Comment by Arslanbekov** [comment]
#8

File: Makefile [hunk 1]
```diff
@@ -1,55 +1,87 @@
...
+
 # strict variables
-APP:=slack-duty-bot
-ROOT_DIR:=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+override ROOT_DIR=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+ifneq (, $(shell which git))
+override MOD_NAME=$(shell git config --get remote.origin.url | cut -c 5- | rev | cut -c 5- | rev | tr : / || consul-initializer)
+endif
+ifeq ($(MOD_NAME),)
+override MOD_NAME=consul-initializer
```
[the remark this answers is not in the document]
Comment (insidieux) [reply Makefile]: Fixed

File: Makefile [hunk 2]
```diff
@@ -1,55 +1,87 @@
...
+endif
+
+APP_NAME?=slack-duty-bot
+
 # strict variables
-APP:=slack-duty-bot
-ROOT_DIR:=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+override ROOT_DIR=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+ifneq (, $(shell which git))
+override MOD_NAME=$(shell git config --get remote.origin.url | cut -c 5- | rev | cut -c 5- | rev | tr : / || consul-initializer)
```
[the remark this answers is not in the document]
Comment (insidieux) [reply Makefile]: Fixed

File: .docker/Dockerfile [hunk 3]
```diff
@@ -1,12 +1,10 @@
...
-COPY .docker/docker-entrypoint /usr/local/bin/
+COPY "${APP_NAME}" /usr/local/bin/slack-duty-bot
 
-RUN apk --no-cache add ca-certificates
-
-ENTRYPOINT ["docker-entrypoint"]
+RUN apk --no-cache add ca-certificates && \
+    chmod +x /usr/local/bin/slack-duty-bot
 
 CMD ["slack-duty-bot"]
```
[the remark this answers is not in the document]
Comment (insidieux) [reply .docker/Dockerfile]: `CMD` is just main process for this docker container
All arguments are passed via 
```bash docker run slack-duty-bot --config.path=%your-path%```

File: main.go [hunk 4]
```diff
@@ -56,60 +69,63 @@ func main() {
...
 	if err := validateArguments(); err != nil {
-		log.Fatalf("Validation arguments error: %+v", err)
+		logrus.Fatalf("Validation arguments error: %+v", err)
 	}
 
 	client := slack.New(viper.GetString("slack.token"))
-	if log.GetLevel() == log.DebugLevel {
+	if logrus.GetLevel() == logrus.DebugLevel {
 		client.SetDebug(true)
+		slack.SetLogger(log.New(&slackLogAdapter{
```
[the remark this answers is not in the document]
Comment (insidieux) [reply main.go]: There is no mistake. Slack logger need to implementation `Output` method, which is not supported by `logrus` package. So, decision is to create adapter and pass it to new std logger.

**Pull request merged**